generator/computers_c.py

```diff
@@ -78,28 +78,68 @@ def write(can, computers, output_path=computer_c_dir_path):
 
             fw('\t}\n}\n\n')
 
+            fw(
+              'CAN_Raw_Bus_T CANlib_GetConceptualBus(CAN_Raw_Bus_T bus) {\n'
+              '\tswitch (bus) {\n'
+            )
+
+            for busnm, rawnm in computer.participation['name']['can'].mapping.items():
+                fw('\t\tcase {}:\n'.format(rawnm) + '\t\t\treturn {};\n'.format(busnm))
+
+            fw('\t\tdefault:\n\t\t\treturn INVALID_BUS;\n')
+
+            fw('\t}\n}\n\n')
+
             for busnm, bus in computer.participation['name']['can'].subscribe.items():
                 fw(
-                    'static void CANlib_update_can_{}(void)'.format(busnm) + '{\n' +
-                    '\tFrame frame;\n'
-                )
-                if any(is_multplxd(msg) for msg in bus):
-                    fw('\tuint64_t bitstring;\n')
-                fw(
-                    '\tCANlib_ReadFrame(&frame, {});\n'.format(busnm) +
-                    '\tswitch(frame.id) {\n'
+                    'static void CANlib_HandleFrame_{}(Timestamped_Frame* ts_frame)'.format(busnm) + '{\n' +
+                        '\tswitch(ts_frame->frame.id) {\n'
                 )
 
                 for msg in bus:
                     msg_handler(msg, busnm, fw)
 
+                fw('\t\tdefault:\n\t\t\treturn;\n')
                 fw('\t}\n')
 
                 fw(
                     '}\n\n'
                 )
 
+            for busnm, bus in computer.participation['name']['can'].subscribe.items():
+                fw(
+                    'static void CANlib_update_can_{}(void)'.format(busnm) + '{\n' +
+                    '\tTimestamped_Frame ts_frame;\n'
+                )
+                if any(is_multplxd(msg) for msg in bus):
+                    fw('\tuint64_t bitstring;\n')
+
+                fw(
+                    '\tif (CANlib_ReadFrame(&(ts_frame.frame), {})) {{\n'.format(busnm) +
```

Comment:
You don't set the timestamp so it's initialized with junk. Here you can set it with get tick